accounts/abi/bind/backends/simulated.go

```diff
@@ -558,6 +606,25 @@ func (b *SimulatedBackend) SendTransaction(ctx context.Context, tx *types.Transa
 	b.mu.Lock()
 	defer b.mu.Unlock()
 
+	// EIP1559 guards
+	eip1559 := b.config.IsEIP1559(b.pendingBlock.Number())
+	eip1559Finalized := b.config.IsEIP1559Finalized(b.pendingBlock.Number())
```

Comment:
As a general comment, I am wondering if we should completely remove transition period related code rather than setting the period to 0. I think it is confusing and most of this code anyway with a value of zero.